docs/guides/batching.md

```diff
@@ -0,0 +1,183 @@
+
+### Meal Prep and Delivery
+
+Before meals are prepped, the meal delivery app will first check how many meals it will need to make for each company and the location of the employees where each meal will be delivered. Since employees may sometimes switch between working at the office, at a co-working space or from home, the app tries to estimate each employee's current location by geolocation of their IP address.
+
+```graphql showLineNumbers
+schema
+  @server(port: 8000, graphiql: true)
+  # highlight-start
+  @upstream(baseURL: "https://geoip-batch.fly.dev", httpCache: true, batch: {delay: 1, maxSize: 100}) {
+  # highlight-end
+  query: Query
+}
+
+type Query {
+  users: [User]! @http(path: "/users")
+}
+
+type User {
+  id: Int!
+  username: String!
+  email: String!
+  phone: String
+  ip: String!
+  # highlight-start
+  country: Country! @http(path: "/batch", query: [{key: "query", value: "{{value.ip}}"}], groupBy: ["query"])
+  # highlight-end
+}
+
+type Country {
+  query: String
+  country: String
+  regionName: String
+  city: String
+  lat: Float
+  lon: Float
+}
+```
+
+A lot of geolocation services support batch requests to save on network round-trips. The sample `graphql` shows how batching could be used to lookup the location of multiple employees using only one batch request with the batch size set to `100` sub-requests.
+
+When you run the following GraphQL query:
+
+```graphql
+{
+  users {
+    id
+    username
+    email
+    phone
+    ip
+    country {
+      query
+      country
+      regionName
+      city
+      lat
+      lon
+    }
+  }
+}
+```
+
+It will produce the following output in Tailcall:
+
+```bash
+2024-01-22T13:57:33Z INFO  tailcall::cli::tc] N + 1: 0
+[2024-01-22T13:57:33Z INFO  tailcall::http] 🚀 Tailcall launched at [127.0.0.1:8000] over HTTP/1.1
+[2024-01-22T13:57:33Z INFO  tailcall::http] 🌍 Playground: http://127.0.0.1:8000
+[2024-01-22T13:58:04Z INFO  tailcall::http::client] GET https://geoip-batch.fly.dev/users HTTP/1.1
+[2024-01-22T13:58:05Z INFO  tailcall::http::client] GET https://geoip-batch.fly.dev/batch?query=100.159.51.104&query=103.72.86.183&query=116.92.198.102&query=117.29.86.254&query=137.235.164.173&query=141.14.53.176&query=163.245.232.27&query=174.238.43.126&query=197.37.13.163&query=205.226.160.3&query=25.207.107.146&query=29.82.54.30&query=43.20.78.113&query=48.30.193.203&query=49.201.206.36&query=51.102.180.216&query=53.240.20.181&query=59.43.194.22&query=71.57.235.192&query=73.15.179.178&query=74.80.53.208&query=75.75.234.243&query=78.170.185.120&query=78.43.74.226&query=82.170.69.15&query=87.213.156.73&query=90.202.216.39&query=91.200.56.127&query=93.246.47.59&query=97.11.116.84 HTTP/1.1
+```
+
+The `/users` endpoint returns a total of 30 users. As you can see in the output, Tailcall constructed a batch request using the IP addresses of all 30 users rather than make 30 individual requests to the geolocation service.
```

Comment:
we need to describe more how is batching is more efficient here and how does it affect backend. Event if it was mentioned before. 
Also point out about the deduplication that happens when batching is enabled

Reply:
@meskill I've reworded.